src/core/transport/chttp2/internal.h

```diff
@@ -0,0 +1,652 @@
+  gpr_uint32 id;
+  /** sops that have passed flow control to be written */
+  grpc_stream_op_buffer sopb;
+  /** how strongly should we indicate closure with the next write */
+  grpc_chttp2_send_closed send_closed;
+} grpc_chttp2_stream_writing;
+
+struct grpc_chttp2_stream_parsing {
+  /** HTTP2 stream id for this stream, or zero if one has not been assigned */
+  gpr_uint32 id;
+  /** has this stream received a close */
+  gpr_uint8 received_close;
+  /** saw a rst_stream */
+  gpr_uint8 saw_rst_stream;
+  /** incoming_window has been reduced by this much during parsing */
+  gpr_uint32 incoming_window_delta;
+  /** window available for peer to send to us */
+  gpr_uint32 incoming_window;
+  /** parsing state for data frames */
+  grpc_chttp2_data_parser data_parser;
+  /** reason give to rst_stream */
+  gpr_uint32 rst_stream_reason;
+  /* amount of window given */
+  gpr_uint64 outgoing_window_update;
+
+  /** incoming metadata */
+  grpc_chttp2_incoming_metadata_buffer incoming_metadata;
+
+  /*
+    grpc_linked_mdelem *incoming_metadata;
+    size_t incoming_metadata_count;
+    size_t incoming_metadata_capacity;
+    grpc_linked_mdelem *old_incoming_metadata;
+    gpr_timespec incoming_deadline;
+  */
+};
+
+struct grpc_chttp2_stream {
+  grpc_chttp2_stream_global global;
+  grpc_chttp2_stream_writing writing;
+  grpc_chttp2_stream_parsing parsing;
+
+  grpc_chttp2_stream_link links[STREAM_LIST_COUNT];
+  gpr_uint8 included[STREAM_LIST_COUNT];
+
+#if 0
+  gpr_uint32 outgoing_window_update;
+  gpr_uint8 cancelled;
+
+  grpc_stream_state callback_state;
+  grpc_stream_op_buffer callback_sopb;
+#endif
+};
+
+/** Transport writing call flow:
+    chttp2_transport.c calls grpc_chttp2_unlocking_check_writes to see if writes
+   are required;
+    if they are, chttp2_transport.c calls grpc_chttp2_perform_writes to do the
+   writes.
+    Once writes have been completed (meaning another write could potentially be
+   started),
+    grpc_chttp2_terminate_writing is called. This will call
+   grpc_chttp2_cleanup_writing, at which
+    point the write phase is complete. */
+
+/** Someone is unlocking the transport mutex: check to see if writes
+    are required, and schedule them if so */
+int grpc_chttp2_unlocking_check_writes(grpc_chttp2_transport_global *global,
+                                       grpc_chttp2_transport_writing *writing);
+void grpc_chttp2_perform_writes(
+    grpc_chttp2_transport_writing *transport_writing, grpc_endpoint *endpoint);
+void grpc_chttp2_terminate_writing(
+    grpc_chttp2_transport_writing *transport_writing, int success);
+void grpc_chttp2_cleanup_writing(grpc_chttp2_transport_global *global,
+                                 grpc_chttp2_transport_writing *writing);
+
+void grpc_chttp2_prepare_to_read(grpc_chttp2_transport_global *global,
+                                 grpc_chttp2_transport_parsing *parsing);
+/** Process one slice of incoming data */
```

Comment:
Would you mind documenting the return value of this function and how it may get used? I think I know, but I want to make sure that my interpretation is clear and correct.